lang/c++/api/NodeConcepts.hh

```diff
@@ -47,33 +47,29 @@ namespace avro {
 
 namespace concepts {
 
-template <typename Attribute>
-struct NoAttribute
-{
+template<typename Attribute>
+struct NoAttribute {
     static const bool hasAttribute = false;
 
     size_t size() const {
         return 0;
     }
 
-    void add( const Attribute &attr) {
+    void add(const Attribute &attr) {
```

Comment:
why not remove attr if it isn't used here?

Reply:
This is because `NoAttribute` is passed to some template which would call `add` on it conditionally. The condition is never met for this specific type and hence this exception is never seen. But the compiler would complain without this function. Not a great design, but we have it.

Reply:
I might be misunderstanding (or maybe my comment wasn't clear).  But it seems like `void add(const Attribute& /*attr*/)` would still compile?

Reply:
Got it and fixed it